lib/Geo/GeoJson.cpp

```diff
@@ -353,40 +355,68 @@ Result parsePolygon(VPackSlice const& vpack, ShapeContainer& region) {
           std::string("Invalid loop in polygon: ").append(error.text()));
     }
 
+    // Note that we are using InitNested for S2Polygon below.
+    // Therefore, we are supposed to deliver all our loops in CCW
+    // convention (aka right hand rule, interiour is to the left of
+    // the polyline).
+    // Since we want to want to allow for loops, whose interiour covers
+    // more than half of the earth, we must not blindly "Normalize"
+    // the loops, as we did in earlier versions, although RFC7946
+    // says "parsers SHOULD NOT reject Polygons that do not follow
+    // the right-hand rule". Since we cannot detect if the outer loop
+    // respects the rule, we cannot reject it. However, for the other
+    // loops, we can be a bit more tolerant: If a subsequent loop is
+    // not contained in the first one (following the right hand rule),
+    // then we can invert it silently and if it is then contained
+    // in the first, we have proper nesting and leave the rest to
+    // InitNested. This is, why we proceed like this:
     S2Loop* loop = loops.back().get();
-    // normalization ensures that point orientation does not matter for Polygon
-    // type the RFC recommends this for better compatibility
-    loop->Normalize();
+    if (legacy) {
+      loop->Normalize();
+    }
 
-    // subsequent loops must be holes within first loop
+    // subsequent loops must be holes within first loop:
     if (loops.size() > 1 && !loops.front()->Contains(loop)) {
-      return Result(TRI_ERROR_BAD_PARAMETER,
-                    "Subsequent loop not a hole in polygon");
+      bool bad = false;
+      if (legacy) {
```

Comment:
```suggestion
      if (legacy || !loops.back()->Contains(loop) {
        bad = true;
      }
      if (!legacy) {
        loop->Invert();
      }
```
maybe something like this to make use of the same statement?

Reply:
and, if so, the same with parseMultiPolygon

Reply:
Sorry, I do not understand this transformation and I believe it is wrong. It is critical here that we first check containment in line 379, and (if not legacy), then first try to invert (which changes the loop in place) and then check again for containment. Your code change would simply check the polygon twice in the non-legacy case, without first inverting it.